pkg/Library/NodeGraph/ConnectionUpdator.js

```diff
@@ -16,12 +16,14 @@ update(inputs, state) {
   if (this.inputsChanged(inputs, state)) {
     const {pipeline, nodeTypes, candidates} = inputs;
     assign(state, {pipeline, candidates});
-    let changed = false;
-    if (this.removePipelineOutdatedConnections(pipeline, candidates)) {
-      changed = true;
-    }
-    if (this.updatePipelineConnections(pipeline, nodeTypes, candidates)) {
-      changed = true;
+    if (candidates) {
```

Comment:
my intention here was to:

if (changed) {
   return {pipeline};
}
i forgot to add this if statement 🤦 , but should we avoid returning, if nothing was updated?

Reply:
Generally it's good to avoid returning. We can fix this in a follow up.